extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -478,27 +478,31 @@ export class SubscriptionService {
   }
 
   /**
-   * Re authorizes a platform
-   * @param {!SubscriptionPlatform} subscriptionPlatform
+   * Reset all platforms and re-fetch entitlements after an
+   * external event (for example a login)
    */
-  reAuthorizePlatform(subscriptionPlatform) {
-    this.platformStore_.reset();
-    this.platformStore_.getAvailablePlatforms()
-        .forEach(platform => {
-          platform.reset();
-          this.platformStore_.resetEntitlementFor(
-              platform.getServiceId());
-          this.fetchEntitlements_(platform);
-        });
+  resetPlatforms() {
+    const serviceIds = this.platformConfig_['services'].map(service =>
+      service['serviceId'] || 'local');
+
+    this.initializePlatformStore_(serviceIds);
```

Comment:
Wouldn't this still lose the references to the platforms which have registered themselves by calling 

https://github.com/ampproject/amphtml/blob/master/extensions/amp-subscriptions/0.1/amp-subscriptions.js#L172